src/user/dto/update-user.dto.ts

```diff
@@ -1,4 +1,39 @@
 import { PartialType } from '@nestjs/mapped-types';
 import { CreateUserDto } from './create-user.dto';
+import {
+  IsEmail,
+  MaxLength,
+  MinLength,
+  Matches,
+  IsOptional,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
 
-export class UpdateUserDto extends PartialType(CreateUserDto) {}
+export class UpdateUserDto extends PartialType(CreateUserDto) {
+  @IsOptional()
+  @ApiProperty({ example: 'user@example.com' })
+  @IsEmail()
+  email: string;
+
+  @ApiProperty({
+    example: 'password123',
+    description:
+      'Password must be at least 6 characters long and no more than 20 characters long! Password must contain at least one digit and  at least one lowercase letter and at least one uppercase letter!',
+  })
+  @IsOptional()
+  @MinLength(6, { message: 'Password must be more than 6 symbols!' })
```

Comment:
Perhaps you could update the message to 'Password must be at least 6 characters,' as the current message isn't quite accurate.

Reply:
typo ) but i'll update

Reply:
updated